src/EditorFeatures/Core/Shared/Tagging/Utilities/TagSpanIntervalTree.cs

```diff
@@ -134,49 +139,62 @@ private void AddTagsForLargeNumberOfSpans(NormalizedSnapshotSpanCollection reque
 
             using var _1 = SegmentedListPool.GetPooledList<ITagSpan<TTag>>(out var tempList);
 
-            AddIntersectingSpans(mergedSpan, tempList);
+            AppendIntersectingSpansInSortedOrder(mergedSpan, tempList);
             if (tempList.Count == 0)
                 return;
 
+            // Note: both 'requstedSpans' and 'tempList' are in sorted order.
+
             using var enumerator = tempList.GetEnumerator();
 
             if (!enumerator.MoveNext())
                 return;
 
             using var _2 = PooledHashSet<ITagSpan<TTag>>.GetInstance(out var hashSet);
 
+            var requestIndex = 0;
             while (true)
             {
-                var requestIndex = 0;
```

Comment:
this was hte mistake when i did the original refactoring here.  by resetting requestIndex to 0 at hte top of hte loop we were effectively never moving that  item forward in the second if-check.

Reply:
You can see here that this was appropriately outside the loop prior to the mistake: https://github.com/dotnet/roslyn/pull/70884/files?diff=split#diff-9d0243eeadbc7ee94e5712ce574223cb596fa3a6110785a8e34e9e90dcc98e55L134